### ISSUE TITLE
Auto-generated addon index files

### ISSUE THREAD

**Issue by jakecraige** [ember-cli/ember-cli#1376]
There's a log of boilerplate right now when creating an addon and with some conventions I don't think it would be necessary. 

taken/modified from @bcardarella [latest PR](https://github.com/bcardarella/ember-cli/blob/bc-addon-blueprint/blueprints/addon/files/index.js)

``` js
'use strict';

var fs   = require('fs');
var path = require('path');

function DefaultAddon(project) {
  this.project = project;
  this.name    = '<%= classifiedPackageName %>';
}

function unwatchedTree(dir) {
  return {
    read:    function() { return dir; },
    cleanup: function() { }
  };
}

DefaulAddon.prototype.treeFor = function treeFor(name) {
  var treePath = path.relative(process.cwd(), __dirname);

   treePath = path.join(treePath, 'app', name);

  if (fs.existsSync(treePath)) {
    return unwatchedTree(treePath);
  }
};

DefaultAddon.prototype.included = function included(app) {
  this.app = app;
};

module.exports = DefaultAddon;
```

You would still need a package.json to specify it's an addon but if you name your folders in a specific way you wouldn't need to define the treeFor or anything else. Addon name could be pulled from package.json as well. 

In addition to this, a default name for the `main` key for ember-addons would be a nice convention to add as well. Easily overridden by specifying it, but I like the defaults.

In summary.. Lets make it more _magical_, we do use Ember.. hah

Thoughts?
